polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -266,9 +264,6 @@ impl PeerData {
 							let candidates =
 								state.advertisements.entry(on_relay_parent).or_default();
 
-							if candidates.len() > max_candidate_depth {
```

Comment:
This error leads to reporting the peer with `COST_UNEXPECTED_MESSAGE`. I think we shold relax it to just ignoring the advertisement.

Pros:
- with the new logic submitting more elements than scheduled is not such a major offence
- old collators won't get punished for not respecting the claim queue

Cons:
- we don't punish spammy collators

Reply:
I didn't pick the best words here but my point was since we guarantee fair share of core time for each parachain we can close our eyes for collators pushing too much advertisements as they don't cause any harm.

Furthermore at the moment we haven't got a proper collator using the claim queue to decide what advertisements to send (right?) so when this change is live a lot of collators will suffer.

Reply:
We should certainly not keep inserting blindly. You are right that it is less of an issue since we are now guaranteeing fairness, so we don't have to be super strict here. In particular given how ineffective reputation changes are in practice, it might make sense to set the limit to something we are 100% no honest node will ever reach, but then make it an immediate disconnect (max punishment).

Is the concern about old collators valid? Why would they be spammy?

Reply:
we should still have some limit here. otherwise we record unlimited advertisements

Reply:
I agree that without proper spam protection a malicious collator could bombard the validator with advertisements. I've restored the limit but instead of `max_candidate_depth` I've put the length of the current assignments which is equal to the claim queue of the assigned core.